__tests__/useSelectedItems.test.tsx

```diff
@@ -0,0 +1,155 @@
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.items).toContainEqual(
+      expect.objectContaining({ ...initialExampleItems[0], isSelected: true }),
+    );
+  });
+
+  it("should render with empty items", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+    }));
+
+    expect(result.current.items).toEqual([]);
+  });
+
+  it("should toggle a single item", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    act(() => {
+      result.current.toggleSingleItem(1);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    act(() => {
+      result.current.toggleSingleItem(2);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+  });
+
+  it("should toggle a single item", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    act(() => {
+      result.current.toggleSingleItem(1);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    act(() => {
+      result.current.toggleSingleItem(2);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+  });
+
+  it("should toggle all items", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[2]),
+    );
+
+    act(() => {
+      result.current.toggleAllItems();
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[2]),
+    );
```

Comment:
I would suggest to "merge" the assertions by directly verifying if the initial items array is equal to the current `selectedItems` state

```suggestion
    expect(result.current.selectedItems).not.toEqual(initialExampleItems);

    act(() => {
      result.current.toggleAllItems();
    });

    expect(result.current.selectedItems).toEqual(initialExampleItems);
```